corehq/apps/reports/tasks.py

```diff
@@ -370,7 +369,28 @@ def _get_export_properties(export):
     return properties
 
 
-def find_question_id(form, value):
+def _get_question_id_for_attachment(form, attachment_name):
+    """
+    Attempts to build and return a question_id from retrieved path list
+    """
+    question_id_components = _find_path_to_question_id(form, attachment_name, use_basename=False)
+
+    # NOTE: until rd-toolkit bug is fixed, search for question_id again looking at basename of attachment_name
+    # See https://dimagi-dev.atlassian.net/browse/SAAS-11792
+    if question_id_components is None:
+        question_id_components = _find_path_to_question_id(form, attachment_name, use_basename=True)
+
+    try:
+        return str('-'.join(question_id_components))
+    except TypeError:
+        return None
```

Comment:
This is probably outside the scope (I understand it's existing code), but this could be cleaned up.
```suggestion
    if question_id_components:
        return '-'.join(question_id_components)
    return None
```

Reply:
@mjriley's suggested code will return `None` (instead of an empty string) if `question_id_components` is an empty list. Not sure if that's desired or not. Would it be better to check for `None` explicitly?
```py
if question_id_components is not None:
```

Reply:
My reason for keeping this the same was to avoid changing existing behavior, but understand that the block:
```
if question_id_components is not None:
    return str('-'.join(question_id_components))
else:
    return None
```
_should_ behave the same as the try except logic.